damnit/backend/extract_data.py

```diff
@@ -197,51 +168,126 @@ def update_db_vars(self):
             ))
         self.kafka_prd.flush()
 
-    def extract_and_ingest(self, proposal, run, cluster=False,
-                           run_data=RunData.ALL, match=(), variables=(), mock=False):
-        if proposal is None:
-            proposal = self.db.metameta['proposal']
+class RunExtractor(Extractor):
+    def __init__(self, proposal, run, cluster=False, run_data=RunData.ALL,
+                 match=(), variables=(), mock=False):
+        super().__init__()
+        self.proposal = proposal
+        self.run = run
+        self.cluster = cluster
+        self.run_data = run_data
+        self.match = match
+        self.variables = variables
+        self.mock = mock
+        self.uuid = str(uuid4())
+        self.running_msg = msg_dict(MsgKind.processing_running, {
```

Comment:
Could we send and display the user it's running under as well?

Reply:
Yup, that makes sense.